pkg/controller/machineset_util.go

```diff
@@ -105,16 +105,37 @@ func (c *controller) syncMachinesNodeTemplates(machineList []*v1alpha1.Machine,
 	return nil
 }
 
+// syncMachinesClassKind updates all machines in the given machineList with the new classKind if required.
+func (c *controller) syncMachinesClassKind(machineList []*v1alpha1.Machine, machineSet *v1alpha1.MachineSet) error {
+
+	controlClient := c.controlMachineClient
+	machineLister := c.machineLister
+
+	for _, machine := range machineList {
+		classKindChanged := copyMachineSetClassKindToMachines(machineSet, machine)
+		// Only sync the machine that doesn't already have the latest nodeTemplate.
+		if classKindChanged {
+			_, err := UpdateMachineWithRetries(controlClient.Machines(machine.Namespace), machineLister, machine.Namespace, machine.Name,
```

Comment:
Ignorable: A `RequeueAfter` equivalent might be more optimal here.

Reply:
Okay, will try to make this change as well.